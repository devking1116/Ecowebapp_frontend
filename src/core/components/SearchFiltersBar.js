/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Button, Menu, Icon } from 'semantic-ui-react';

type Props = {
  onSearchClick: () => void,
  children?: any,
};

/**
 * The search filters header bar.
 */
export default function SearchFiltersBar({ onSearchClick, children, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      menu: {
        marginBottom: 0,
      },
    },
  });

  let preTextChildren = [];
  let textChild = null;
  let postTextChildren = [];
  React.Children.forEach(children, (child) => {
    if(textChild)                       postTextChildren.push(child);
    else if(typeof child === 'string')  textChild = child;
    else                                preTextChildren.push(child);
  });

  return (
    <Menu style={styles.menu} borderless {...otherProps}>
    {preTextChildren.map((child, index) =>
      <Menu.Item key={index}>
        {child}
      </Menu.Item>
    )}
      <Menu.Menu position="right">
        <Menu.Item>
          {textChild}
        </Menu.Item>
      {postTextChildren.map((child, index) =>
        <Menu.Item key={index}>
          {child}
        </Menu.Item>
      )}
        <Menu.Item>
          <Button color="blue" onClick={onSearchClick}>
            <Icon name="search" />
          </Button>
        </Menu.Item>
      </Menu.Menu>
    </Menu>
  );
}
