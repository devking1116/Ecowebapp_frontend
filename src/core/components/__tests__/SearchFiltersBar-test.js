import React from 'react';
import renderer from 'react-test-renderer';

import SearchFiltersBar from '../SearchFiltersBar';

test('it displays its children in a secondary Menu.', () => {
  const component = renderer.create(
    <SearchFiltersBar>
      <div>Test filter 0</div>
      Test text:
      <div>Test filter 1</div>
      <div>Test filter 2</div>
    </SearchFiltersBar>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('it calls the update seatch function when search button is clicked.', () => {
  const searchFn = jest.fn();
  const component = renderer.create(
    <SearchFiltersBar onSearchClick={searchFn}>
      <div>Test filter 0</div>
      Test text:
      <div>Test filter 1</div>
      <div>Test filter 2</div>
    </SearchFiltersBar>
  );
  const tree = component.toJSON();
  const searchButton = tree.children[1].children[3].children[0];

  searchButton.props.onClick();
  expect(searchFn).toHaveBeenCalledTimes(1);
});
