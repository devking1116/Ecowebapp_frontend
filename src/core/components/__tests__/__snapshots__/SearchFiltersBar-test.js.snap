exports[`test it displays its children in a secondary Menu. 1`] = `
<div
  className="ui borderless menu"
  style={
    Object {
      "marginBottom": 0,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <div>
      Test filter 0
    </div>
  </div>
  <div
    className="right menu">
    <div
      className="item"
      onClick={[Function]}>
      Test text:
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <div>
        Test filter 1
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <div>
        Test filter 2
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}>
      <button
        className="ui blue button"
        onClick={[Function]}
        tabIndex={undefined}>
        <i
          aria-hidden="true"
          className="search icon" />
      </button>
    </div>
  </div>
</div>
`;
